lang/java/compiler/src/main/java/org/apache/avro/compiler/idl/SchemaResolver.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2015 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.avro.compiler.idl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.avro.JsonProperties;
+import org.apache.avro.LogicalType;
+import org.apache.avro.Protocol;
+import org.apache.avro.Schema;
+import org.codehaus.jackson.JsonNode;
+
+/**
+ *
+ * @author zoly
+ */
+final class SchemaResolver {
+
+    private SchemaResolver() { }
+
+    private static final String UR_SCHEMA_ATTR = "org.apache.avro.compiler.idl.unresolved.name";
+
+    static Schema unresolvedSchema(final String name) {
+        Schema schema = Schema.createRecord("UnresolvedSchema", "unresolved schema",
+                "org.apache.avro.compiler", false, Collections.EMPTY_LIST);
+        schema.addProp(UR_SCHEMA_ATTR, name);
+        return schema;
+    }
+
+    static boolean isUnresolvedSchema(final Schema schema) {
```

Comment:
See: http://checkstyle.sourceforge.net/config_misc.html#FinalParameters

This is another check style rule I use in my code, it is a useful rule and it prevented a few bugs in my past.